benchto-driver/src/main/java/io/trino/benchto/driver/execution/BenchmarkExecutionDriver.java

```diff
@@ -108,18 +113,25 @@ public BenchmarkExecutionResult execute(Benchmark benchmark, int benchmarkOrdina
         }
     }
 
+    private BenchmarkExecutionResult warmupBenchmark(Benchmark benchmark, Optional<ZonedDateTime> executionTimeLimit)
+    {
+        BenchmarkExecutionResultBuilder resultBuilder = new BenchmarkExecutionResultBuilder(benchmark)
+                .withExecutions(List.of());
+        List<QueryExecutionResult> executions = executeQueries(benchmark, benchmark.getPrewarmRuns(), true, executionTimeLimit);
+        String comparisonFailures = getComparisonFailuresString(executions);
+        if (!comparisonFailures.isEmpty()) {
+            resultBuilder.withUnexpectedException(new RuntimeException(format("Query result comparison failed for queries: %s", comparisonFailures)));
+        }
+        return resultBuilder.build();
+    }
+
     private BenchmarkExecutionResult executeBenchmark(Benchmark benchmark, Optional<ZonedDateTime> executionTimeLimit)
     {
         BenchmarkExecutionResultBuilder resultBuilder = new BenchmarkExecutionResultBuilder(benchmark);
         List<QueryExecutionResult> executions;
         try {
             executions = executeQueries(benchmark, benchmark.getPrewarmRuns(), true, executionTimeLimit);
```

Comment:
```suggestion
    private BenchmarkExecutionResultBuilder warmupBenchmark(Benchmark benchmark, Optional<ZonedDateTime> executionTimeLimit)
    {
        BenchmarkExecutionResultBuilder resultBuilder = new BenchmarkExecutionResultBuilder(benchmark)
                .withExecutions(List.of());
        List<QueryExecutionResult> executions = executeQueries(benchmark, benchmark.getPrewarmRuns(), true, executionTimeLimit);
        String comparisonFailures = getComparisonFailuresString(executions);
        if (!comparisonFailures.isEmpty()) {
            resultBuilder.withUnexpectedException(new RuntimeException(format("Query result comparison failed for queries: %s", comparisonFailures)));
        }
        return resultBuilder;
    }

    private BenchmarkExecutionResultBuilder executeBenchmark(Benchmark benchmark, Optional<ZonedDateTime> executionTimeLimit)
    {
        BenchmarkExecutionResultBuilder resultBuilder = warmupBenchmark(benchmark, executionTimeLimit);
```

Reply:
What's the advantage of that?

Reply:
This way the warmup code can be shared by both code paths, which should help with them it in sync.

Reply:
Note that I couldn't complete my suggestion because GitHub doesn't allow suggestions on deleted lines.

Reply:
It's not that simple to reuse `warmupBenchmark` because of how it handles failures.